src/system_data.jl

```diff
@@ -17,7 +17,7 @@ const SERIALIZATION_METADATA_KEY = "__serialization_metadata__"
 
 Container for system components and time series data
 """
-mutable struct SystemData <: InfrastructureSystemsType
+mutable struct SystemData <: ComponentContainer
```

Comment:
`ComponentContainer <: InfrastructureSystemsType` so this is backwards compatible